doc/user/builders-commands.xml

```diff
@@ -142,37 +142,49 @@ foo.in
 
   <para>
 
-  Note that &cv-link-SOURCE; and &cv-link-TARGET; are expanded
-  in the source and target as well, so you can write:
+  &cv-link-SOURCE; and &cv-link-TARGET; are expanded
+  in the source and target as well:
 
   </para>
 
-  <scons_example name="builderscommands_ex3">
-     <file name="SConstruct" printme="1">
-env.Command('${SOURCE.basename}.out', 'foo.in', build)
-     </file>
-  </scons_example>
-
-  <para>
-
-  which does the same thing as the previous example, but allows you
-  to avoid repeating yourself.
-
-  </para>
+  <!-- NOTE: this used to be an scons_example, but was not complete and
+       didn't have a matching scons_output, which meant problems were not
+       detected. The style of this line is now reused in the last example
+       of the section to make sure it's actually tested
+       (see issue #2905 - which was closed prematurely).
+  -->
+  <sconstruct>
+env.Command('${SOURCE.base}.out', File('foo.in'), build)
+  </sconstruct>
 
   <para>
 
-  It may be helpful to use the <parameter>action</parameter>
-  keyword to specify the action, is this makes things more clear
-  to the reader:
+  Which does the same thing as the previous example, but allows you
+  to write a more generic rule for transforming the source filename
+  to the target filename, since unlike regular Builders,
+  &Command; does not have any built-in rules for that.
 
   </para>
 
-  <scons_example name="builderscommands_ex4">
-     <file name="SConstruct" printme="1">
-env.Command('${SOURCE.basename}.out', 'foo.in', action=build)
-     </file>
-  </scons_example>
+  <tip><para>
+
+  The example uses a <firstterm>Node special attribute</firstterm>
```

Comment:
Is this only true for `Command()` and not true for a builder?

Reply:
problem has to do with when evaluation happens, so same for all builders.

Reply:
Hmmm, I forgot DocBook actually has a `<sidebar>` element. However, I like the rendering of `<tip>` better so leaving that, but adding a title to it.

Reply:
So I'd consider this a bug, but of course documenting it will hopefully keep people from running into that bug.
Is there anyway to indicate this in the docs? something like "currently...., but we have an issue filed to make this work so it may be resolved at some time in the future" ?

Reply:
The word "currently" does appear in the prose; I didn't intend to say more in the "friendlier" user guide.  The manpage doesn't mention this problem at all (was only trying to respond to the confusion that came from someone reading the UG), but we could add something there, maybe?